app/lib/common/dialogs/deactivation_confirmation.dart

```diff
@@ -83,66 +85,48 @@ void deactivationConfirmationDialog(BuildContext context, WidgetRef ref) {
-                ),
-              );
-              final sdk = await ref.read(sdkProvider.future);
-              try {
-                if (!await sdk.deactivateAndDestroyCurrentClient(
-                  passwordController.text,
-                )) {
-                  if (!context.mounted) return;
-                  throw L10n.of(context).deactivationAndRemovingFailed;
-                }
-                // ignore: use_build_context_synchronously
-                if (!context.mounted) {
-                  return;
-                }
-                // remove pop up
-                Navigator.of(context, rootNavigator: true).pop();
-                // remove ourselves
-                Navigator.of(context, rootNavigator: true).pop();
-                context.goNamed(Routes.main.name);
-              } catch (err) {
-                // We are doing as expected, but the lints triggers.
-                // ignore: use_build_context_synchronously
-                if (!context.mounted) {
-                  return;
-                }
-
-                showAdaptiveDialog(
-                  context: context,
-                  builder: (context) => DefaultDialog(
-                    title: Text(
-                      '${L10n.of(context).deactivatingFailed}: \n $err"',
-                      style: Theme.of(context).textTheme.titleSmall,
-                    ),
-                    actions: <Widget>[
-                      ElevatedButton(
-                        onPressed: () =>
-                            Navigator.of(context, rootNavigator: true).pop(),
-                        child: Text(L10n.of(context).close),
-                      ),
-                    ],
-                  ),
-                );
-              }
-            },
-            child: Text(
-              L10n.of(context).deactivate,
+            onPressed: () async => _onConfirm(
+              context,
+              ref,
+              passwordController.text,
             ),
+            child: Text(L10n.of(context).deactivate),
           ),
         ],
       );
     },
   );
 }
+
+Future<void> _onConfirm(
+  BuildContext context,
+  WidgetRef ref,
+  String password,
+) async {
+  EasyLoading.show(
+    status: L10n.of(context).deactivatingYourAccount,
+    dismissOnTap: false,
+  );
+  final sdk = await ref.read(sdkProvider.future);
+  try {
+    if (!await sdk.deactivateAndDestroyCurrentClient(password)) {
+      EasyLoading.dismiss();
+      if (!context.mounted) return;
+      EasyLoading.showError(
+        L10n.of(context).deactivationAndRemovingFailed,
+        duration: const Duration(seconds: 3),
+      );
+      return;
+    }
+    EasyLoading.dismiss();
+    if (!context.mounted) return;
+    context.goNamed(Routes.main.name);
+  } catch (err) {
+    EasyLoading.dismiss();
+    if (!context.mounted) return;
+    EasyLoading.showError(
+      '${L10n.of(context).deactivatingFailed}: \n $err"',
+      duration: const Duration(seconds: 3),
+    );
+  }
```

Comment:
No reason to dismiss if you are using a `showError` right after. Additionally, to avoid having that many checks for context.mounted and return you can just keep the l10n around and do the other stuff regardless. Makes things a bit easier...

```suggestion
  final lang = L10n.of(context);
  EasyLoading.show(
    status: lang.deactivatingYourAccount,
    dismissOnTap: false,
  );
  final sdk = await ref.read(sdkProvider.future);
  try {
    if (!await sdk.deactivateAndDestroyCurrentClient(password)) {
      EasyLoading.showError(
        lang.deactivationAndRemovingFailed,
        duration: const Duration(seconds: 3),
      );
    } else {
      if (!context.mounted) return;
      context.goNamed(Routes.main.name);
    }
  } catch (err) {
    EasyLoading.showError(
      '${lang.deactivatingFailed}: \n $err"',
      duration: const Duration(seconds: 3),
    );
  }
```

Reply:
Fixed by https://github.com/acterglobal/a3/pull/1580/commits/e46df3fe7c86d70f46d4f2893e30dba77db5953d